proj/src/main/webapp/script.js

```diff
@@ -50,17 +50,85 @@ function errorCallback(err){
 function getLocation() {
     if(navigator.geolocation){
         navigator.geolocation.getCurrentPosition(successCallback,errorCallback,{timeout:10000, enableHighAccuracy:false});
-    }
-    else{
+    } else{
         console.log("Error. Geolocation not supported or not enabled");
     }
     return;
 }
 
+/*
+    Setting password 
+*/ 
+
+// Check if the password meets the password requirement 
+$('#pass').on('keyup', function(){
+    const password = document.getElementById("pass");
+    var goodPassword = false;
+    passwordLength(password, goodPassword);
+    containsNumber(password, goodPassword);
+    containsSymbol(password, goodPassword);
+    // enables sign up button if the pw requirements are met
+    if (goodPassword === true) {
+        $('#btnSignUp').prop('disabled', false);
+    } else{
+        $('#btnSignUp').prop('disabled', true);
+    }
+});
+
+function passwordLength(password){
+    if (password.value.length > 7){
+        $('#pwLength').addClass('alert-success');
+        goodPassword = true; 
+    } else{
+        $('#pwLength').removeClass('alert-success');
+        goodPassword = false;
+    }
+}
+
+function containsNumber(password){
+    var number = /[0-9]/;
+    if(password.value.match(number)){
+        $('#pwNumber').addClass('alert-success');
+        goodPassword = true;
+    } else{
+        $('#pwNumber').removeClass('alert-success');
+        goodPassword = false;
+    }
+}
+
+function containsSymbol(password){
+    var symbol = /[$-/:-?{-~!"^_`\[\]]/;
+    // const password = document.getElementById("pass");
+    if(password.value.match(symbol)){
+        $('#pwSymbol').addClass('alert-success');
+        goodPassword = true;
+    } else{
+        $('#pwSymbol').removeClass('alert-success');
+        goodPassword = false;
+    }
+}
+
+// Check for password confirmation
+$('#pass, #passconf').on('keyup', function(){
+    if ((pass.value.length != 0) && (passconf.value.length != 0)){
+        if ($('#pass').val() == $('#passconf').val()){
+            $('#btnSignUp').prop('disabled', false);
+            $('#pass').css('border-bottom','2px solid #d1b280');
+            $('#passconf').css('border-bottom','2px solid #d1b280');
+        } else{
+            // disable sign up button
+            $('#btnSignUp').prop('disabled', true);
```

Comment:
I think the sign up button should also be disabled if the passconf button's length is 0. Here if you have two matching passwords then erase all of passconf at once, it might jump to not doing this check and keep the button enabled.

Reply:
there is a required html tag on both inputs so the form cannot be submitted if the passconf's length is 0

Reply:
also firebase seems to throw their own error if the password is too short

Reply:
also the function that enables the sign up button also verifies that both fields are not empty

Reply:
makes sense, especially for errors like invalid password or email already in use